### PR TITLE
Move docker and dockerd main packages in directories with same name, under `cmd/`

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard jhoward@microsoft.com

@dnephin @tiborvass @icecrime 

Ugh, one small hiccup from the daemon split PR - it makes dockerd un-debuggable using LiteIDE - it (correctly) assumes that the name of the binary to be launched is the name of the directory, which was still docker. Hence it tries to launch the client binary (if in path), rather than dockerd with GDB attached. 

Simple fix - rename the docker directory to dockerd, and a small tweak to the build files to use the newly named directory.

### CODE REVIEW

**Comment by tiborvass** [comment]
Seems like a bug in LiteIDE to me :S

**Comment by jhowardmsft** [comment]
@tiborvass Urrm, that really doesn't help me... Are you saying does or does not LGTU?

**Comment by tiborvass** [comment]
@jhowardmsft ok I think I understand the problem now. It makes sense to me to separate the two binaries in two directories. I wonder if we could move `./client` to `cmd/docker` and `./docker` to  `cmd/dockerd` to make things cleaner.

**Comment by vdemeester** [comment]
@tiborvass I like the `cmd/docker` and `cmd/dockerd` package idea :angel: :+1:

**Comment by jhowardmsft** [comment]
Are you really sure you want that extra level of redirection??? Not at the top-level?

**Comment by tiborvass** [comment]
Yes it's an unwritten convention in go repos that everything in cmd/ has
folders with package main in them.

**Comment by jhowardmsft** [comment]
Ok... Update coming soon

**Comment by jhowardmsft** [comment]
Updated. Think I got everything with the client changes too. CI will tell us for sure.

**Comment by jhowardmsft** [comment]
Ugh. Failed TP5 with a known flakey test. Otherwise all looks good. Will restart it anyway.

```
13:14:53 ----------------------------------------------------------------------
13:14:53 FAIL: docker_cli_restart_test.go:222: DockerSuite.TestRestartContainerwithRestartPolicy
13:14:53 
13:14:53 docker_cli_restart_test.go:229:
13:14:53     c.Assert(err, checker.IsNil)
13:14:53 ... value *errors.errorString = &errors.errorString{s:"condition \"\"false true\" == \"false false\"\" not true in time"} ("condition \"\"false true\" == \"false false\"\" not true in time")
13:14:53 
13:15:15 
13:15:15 ----------------------------------------------------------------------
```

**Comment by tiborvass** [comment]
LGTM

**Comment by dnephin** [comment]
I think you'll need to change the paths in `hack/.vendor-helpers.sh:clean()` again.

Otherwise sounds good. I was thinking of doing this, but didn't want to move the files at the same time I was making changes to them.

**Comment by tiborvass** [comment]
And while we're at it, we can remove dockerinit too from there :)

**Comment by jhowardmsft** [comment]
@dnephin Good catch, updated.

@tiborvass git grep dockerinit only place found is in `CHANGES.md`. So no idea what I'm supposed to be removing....

**Comment by LK4D4** [comment]
@jhowardmsft Need rebase :/
I want this hard too

File: .gitignore [hunk 1]
```diff
@@ -12,7 +12,9 @@
 .gopath/
 autogen/
 bundles/
-docker/docker
+binary/
```
Comment (LK4D4) [inline .gitignore]: Is this necessary?

Comment (jhowardmsft) [inline .gitignore]: I build locally to binary\ after the split, rather than having my path set to cmd/docker and cmd/dockerd. Only reason why here.

**Comment by jhowardmsft** [comment]
Oh man. Yes, rebasing.

**Comment by jhowardmsft** [comment]
Rebased

File: .gitignore (hunk 1 above)
Comment (LK4D4) [inline .gitignore]: I'm not really a fan of custom user stuff in `.gitignore`. You can put it in your local `~/.gitignore`.

Comment (jhowardmsft) [inline .gitignore]: Removed

**Comment by jhowardmsft** [comment]
@LK4D4 LGTU now?

**Comment by LK4D4** [comment]
@jhowardmsft oh, it needs rebase again
I promise to merge right after.
LGTM

**Comment by jhowardmsft** [comment]
Uuuuuuuuuuuuuuuuuuurgh. Rebasing...

**Comment by jhowardmsft** [comment]
@LK4D4 OK, rebased.

**Comment by LK4D4** [comment]
hmmm, actually there is janky error. and because it related to reexec it might be a real error.

**Comment by jhowardmsft** [comment]
Investigating. Removing merge tag for now.

**Comment by LK4D4** [comment]
It works for me locally, though. Maybe restart will fix problem.

**Comment by jhowardmsft** [comment]
Yeah, I restarted. I can't see anything in the changes that would cause this.

**Comment by tianon** [comment]
Nice!  I'm a fan of the `cmd/...` approach! :heart:

LGTM

**Comment by LK4D4** [comment]
\o/

**Pull request merged**